Lib/test/test_deque.py

```diff
@@ -892,6 +892,19 @@ def __iter__(self):
         d1 == d2   # not clear if this is supposed to be True or False,
                    # but it used to give a SystemError
 
+    @support.cpython_only
+    def test_bug_31608(self):
+        # The interpreter used to crash in specific cases where a deque
+        # subclass returned a non-deque.
+        class X(deque):
+            pass
+        d = X()
+        def bad___new__(cls, *args, **kwargs):
+            return 42
+        X.__new__ = bad___new__
+        with self.assertRaises(TypeError):
+            d * 42  # shouldn't crash
```

Comment:
What about concatenation?

Reply:
The C changes are only to `deque_copy()`, so i thought that testing also the concatenation is redundant.
Should i add it to the test anyway?